l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -183,68 +130,34 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         uint256 _l2TxGasPerPubdataByte,
         address _refundRecipient
     ) public payable nonReentrant returns (bytes32 l2TxHash) {
-        require(_amount != 0, "2T"); // empty deposit amount
-        uint256 amount = _depositFunds(msg.sender, IERC20(_l1Token), _amount);
-        require(amount == _amount, "1T"); // The token has non-standard transfer logic
-
-        bytes memory l2TxCalldata = _getDepositL2Calldata(msg.sender, _l2Receiver, _l1Token, amount);
-        // If the refund recipient is not specified, the refund will be sent to the sender of the transaction.
-        // Otherwise, the refund will be sent to the specified address.
-        // If the recipient is a contract on L1, the address alias will be applied.
-        address refundRecipient = _refundRecipient;
-        if (_refundRecipient == address(0)) {
-            refundRecipient = msg.sender != tx.origin ? AddressAliasHelper.applyL1ToL2Alias(msg.sender) : msg.sender;
-        }
-        l2TxHash = zkSync.requestL2Transaction{value: msg.value}(
-            l2Bridge,
-            0, // L2 msg.value
-            l2TxCalldata,
+        require(_amount != 0, "0T"); // empty deposit
+        uint256 amount = _depositFundsToSharedBridge(msg.sender, IERC20(_l1Token), _amount);
+        require(amount == _amount, "3T"); // The token has non-standard transfer logic
```

Comment:
This is a good start to detect non-standard transfer logic, but, what is our plan in case of tokens like USDC, USDT that are upgradable and can change their logic to **include** non-standard transfer logic after calling the function `deposit` anytime they want?

Reply:
The concept of bridging to another chain breaks down if we don't have the required amount of tokens. For these we will have to use contracts that are aware that they are bridged, i.e. contracts similar to L2StandardToken. Then we don't hold value in the bridge, so we don't have to receive any amount.